backend/game/player/ai_players/master_ai_player.py

```diff
@@ -46,3 +46,8 @@ def attackable_players(self):
             if player.location != self.location:
                 attackables.append(player)
         return attackables
+
+    def should_heal_self(self):
+        if (self.maximum_health - self.selfcurrent_health) < 5:
```

Comment:
`self.selfcurrent_health` should be `self.current_health`

I think the less than  should actually be  a greater than. Imagine your current health is 9. Then `(self.maximum_health - self.selfcurrent_health)` = 1 and you would try to heal. But then if your current health is 4 then `(self.maximum_health - self.selfcurrent_health)` = 6 and you would stop trying to heal